### PR TITLE
Add themeable cobbangle to the measurements tools

### PR DESCRIPTION
**Pull request by wayfarer3130**
This change adds support for a Cobb Angle tool.  The use of the tool can be enabled by including the theme=extraTools parameter on the URL, which loads the configuration settings to add in the cobb angle tool.  There is also documentation added to specify how to change the toolbar menus via configuration.

### CODE REVIEW

**Comment by wayfarer3130** [comment]
@ladeirarodolfo  - this change is for part of the work we are doing right now, and is tested against the static wado stow rs endpoints.

**Review by ladeirarodolfo — CHANGES_REQUESTED** [review]

File: platform/viewer/public/config/dicomweb-server.js [hunk 1]
```diff
@@ -12,12 +12,12 @@ window.config = {
       sourceName: 'dicomweb',
       configuration: {
         name: 'DCM4CHEE',
-        wadoUriRoot: 'http://localhost:5985',
-        qidoRoot: 'http://localhost:5985',
-        wadoRoot: 'http://localhost:5985',
+        wadoUriRoot: 'http://localhost:5984',
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/viewer/public/config/dicomweb-server.js:15]: This review is against another PR, not the core v3-stable, and this is a revert of a change in the other review.

File: platform/viewer/src/index.js [hunk 2]
```diff
@@ -37,8 +38,17 @@ const appProps = {
...
 
-/** Create App */
-const app = React.createElement(App, appProps, null);
+/**
+ * Load the default theme settings,
+ * and then render the app in the then block so that themes get
+ * loaded early enough to modify the initial render.
+ */
+const defaultTheme = config && config.defaultTheme || 'theme';
+loadSearchConfigPoint(defaultTheme, '/theme', 'theme').then(configsLoaded => {
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/viewer/src/index.js:47]: Yes, adding.

File: platform/viewer/public/config/dicomweb-server.js [hunk 3]
```diff
@@ -12,12 +12,12 @@ window.config = {
...
+        qidoRoot: 'http://localhost:5984',
+        wadoRoot: 'http://localhost:5984',
         qidoSupportsIncludeField: true,
         supportsReject: true,
-        imageRendering: 'wadouri',
+        imageRendering: 'wadors,wadouri',
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/viewer/public/config/dicomweb-server.js:20]: Reverting this one.

File: extensions/cornerstone/src/OHIFCornerstoneViewport.js [hunk 4]
```diff
@@ -236,9 +236,14 @@ function OHIFCornerstoneViewport({
         isOverlayVisible={true}
         loadingIndicatorComponent={ViewportLoadingIndicator}
         viewportOverlayComponent={props => {
+          const { imageIndex } = props;
+          const instances = displaySet.images || displaySet.other;
+          const image = instances && imageIndex > 0 &&
+            imageIndex <= instances.length && instances[imageIndex - 1];
+          const viewportProps = { ...props, cine, isPlaying, frameRate, image };
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply extensions/cornerstone/src/OHIFCornerstoneViewport.js:243]: It isn't known whether or not those are being used.  In fact, the default rendering does use them.

File: platform/ui/src/components/ViewportOverlay/index.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+ *
+ * @param {*} config is a configuration object that defines four lists of elements,
+ * one topLeft, topRight, bottomLeft, bottomRight contents.
+ * @param {*} extensionManager is used to load the image data.
+ * @returns
+ */
+const generateFromConfig = (config) => {
+  const { topLeft, topRight, bottomLeft, bottomRight, itemGenerator } = config;
+
+  return (props) => {
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/ui/src/components/ViewportOverlay/index.js:17]: Fixing

File: platform/ui/src/components/ViewportOverlay/index.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    const bottomLeftClass = 'bottom-viewport left-viewport';
+    const overlay = 'absolute pointer-events-none';
+
+    if (!imageId) {
+      return null;
+    }
+
+    return (
+      <div className="text-primary-light">
+        <div
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/ui/src/components/ViewportOverlay/index.js:31]: Thank you, implemented that in the viewport overlay PR.

**Pull request closed without merging**